linkerd/app/outbound/src/ingress.rs

```diff
@@ -73,42 +73,48 @@ where
         .into_inner();
 
     svc::stack(http)
-        .push_on_response(svc::MapErrLayer::new(Into::into))
-        .check_new_service::<http::Logical, http::Request<_>>()
+        .push_on_response(
+            svc::layers()
+                .push(http::BoxRequest::layer())
+                .push(svc::MapErrLayer::new(Into::into)),
+        )
+        // Lookup the profile for the outbound HTTP target, if appropriate.
+        //
+        // This service is buffered because it needs to initialize the profile
+        // resolution and a failfast is instrumented in case it becomes
+        // unavailable
+        // When this service is in failfast, ensure that we drive the
+        // inner service to readiness even if new requests aren't
+        // received.
         .push_map_target(http::Logical::from)
         .push(profiles::discover::layer(
             profiles,
             AllowHttpProfile(allow_discovery),
         ))
-        .check_new_service::<Target, http::Request<_>>()
         .push_on_response(
             svc::layers()
-                .push(svc::FailFast::layer("Logical", dispatch_timeout))
+                .push(svc::layer::mk(svc::SpawnReady::new))
+                .push(svc::FailFast::layer("HTTP Logical", dispatch_timeout))
```

Comment:
TIOLI: might be worth having a "push spawn ready and failfast in one method" helper since we now want to pair them in most places. might also help avoid accidentally adding new failfasts without a spawn ready?

Reply:
Agreed, but we don't have a nameable spawn ready layer at the moment.